torch/csrc/jit/fuser/codegen.cpp

```diff
@@ -224,8 +218,33 @@ static std::string encodeRHS(const Node* n) {
 
   TemplateEnv env;
   size_t i = 0;
-  auto outtype =
-      n->output()->type()->expect<c10::DimensionedTensorType const>()->scalarType();
+  auto outtype = n->output()
+                     ->type()
+                     ->expect<c10::DimensionedTensorType const>()
+                     ->scalarType();
+
+  // special case for clamp fusion on missing min/max inputs
+  if (n->kind() == aten::clamp) {
+    const auto min = n->input(1);
+    const auto max = n->input(2);
+    env.s(std::to_string(i++), valueName(n->input(0)));
+
+    if (!min->node()->mustBeNone() && !max->node()->mustBeNone()) {
+      env.s(std::to_string(i++), valueName(min));
+      env.s(std::to_string(i++), valueName(max));
+      return format("(${0}<${1}?${1}:(${0}>${2}?${2}:${0}))", env);
+    } else if (min->node()->mustBeNone()) {
+      env.s(std::to_string(i++), valueName(max));
+      return format("(${0}<${1}?${0}:${1})", env);
```

Comment:
This won't correctly propagate nan's? I.e. if input {0} is none, comparison will evaluate to false and {1} which is a limit will be output. Would make sense to add tests for nan propagation.